lib/src/controls/form/password_box.dart

```diff
@@ -0,0 +1,284 @@
+
+  /// The text shown when the text box is empty
+  ///
+  /// See also:
+  ///
+  ///  * [TextBox.placeholder]
+  final String? placeholder;
+
+  /// The style of [placeholder]
+  ///
+  /// See also:
+  ///
+  ///  * [TextBox.placeholderStyle]
+  final TextStyle? placeholderStyle;
+
+  /// {@macro flutter.widgets.editableText.cursorWidth}
+  final double cursorWidth;
+
+  /// {@macro flutter.widgets.editableText.cursorRadius}
+  final Radius cursorRadius;
+
+  /// {@macro flutter.widgets.editableText.cursorHeight}
+  final double? cursorHeight;
+
+  /// The color of the cursor.
+  ///
+  /// The cursor indicates the current location of text insertion point in
+  /// the field.
+  final Color? cursorColor;
+
+  /// {@macro flutter.widgets.editableText.showCursor}
+  final bool? showCursor;
+
+  /// The highlight color of the text box.
+  ///
+  /// If [foregroundDecoration] is provided, this must not be provided.
+  ///
+  /// See also:
+  ///  * [unfocusedColor], displayed when the field is not focused
+  final Color? highlightColor;
+
+  /// {@macro flutter.widgets.editableText.readOnly}
+  final bool readOnly;
+
+  /// {@macro flutter.widgets.editableText.obscuringCharacter}
+  final String obscuringCharacter;
+
+  /// Creates a password box
+  const PasswordBox({
+    super.key,
+    this.controller,
+    this.onEditingComplete,
+    this.onSubmitted,
+    this.focusNode,
+    this.enabled = true,
+    this.placeholder,
+    this.revealMode = PasswordRevealMode.peek,
+    this.autofocus = false,
+    this.leadingIcon,
+    this.placeholderStyle,
+    this.cursorWidth = 1.5,
+    this.cursorRadius = const Radius.circular(2.0),
+    this.cursorHeight,
+    this.cursorColor,
+    this.showCursor,
+    this.highlightColor,
+    this.readOnly = false,
+    this.obscuringCharacter = '•',
+  });
+
+  @override
+  State<PasswordBox> createState() => _PasswordBoxState();
+}
+
+class _PasswordBoxState extends State<PasswordBox> {
+  bool peek = false;
+  bool focusCanPeek = true;
+  bool textCanPeek = false;
+
+  TextEditingController? _internalController;
+  TextEditingController? get controller =>
+      widget.controller ?? _internalController;
+
+  FocusNode? _internalNode;
+  FocusNode? get focusNode => widget.focusNode ?? _internalNode;
+
+  bool get _isVisible =>
+      widget.revealMode == PasswordRevealMode.visible ||
+      ((widget.revealMode == PasswordRevealMode.peek ||
+              widget.revealMode == PasswordRevealMode.peekAlways) &&
+          peek);
+
+  bool get _canPeek =>
+      (widget.revealMode == PasswordRevealMode.peekAlways && textCanPeek) ||
+      (widget.revealMode == PasswordRevealMode.peek &&
+          focusCanPeek &&
+          textCanPeek);
+
+  @override
+  void initState() {
+    if (widget.focusNode == null) {
+      _internalNode = FocusNode(debugLabel: '${widget.runtimeType}');
+    }
+    if (widget.controller == null) {
+      _internalController = TextEditingController();
+    }
```

Comment:
We also must do this check on `didUpdateWidget`. It is a common case when debugging to attach a controller/node after the widget was already built (hot reload)